### PR TITLE
Appease ShellCheck

### PR DESCRIPTION
**Pull request by sellout**
ShellCheck has started failing in CI. My guess is that the ubuntu-latest runner
now uses a `/bin/sh` that is actually Bash.

- excludes checksec.sh, as it’s not our code
- fixes lints in other files

### CODE REVIEW

File: contrib/docker/entrypoint.sh [hunk 1]
```diff
@@ -55,4 +55,4 @@ fi
 zcash-fetch-params
 touch .zcash/zcash.conf
 echo "Starting: ${ZCASHD_CMD}"
-eval exec "${ZCASHD_CMD}" "${@}"
+exec "${ZCASHD_CMD}" "${@}"
```
Comment (sellout) [inline contrib/docker/entrypoint.sh:58]: This change is wrong.

Comment (sellout) [reply contrib/docker/entrypoint.sh:58]: I think `eval "exec ${ZCASHD_CMD} ${*}"` will do the right thing. Going to test it, but if you know or have a recommendation …

Comment (nuttycom) [reply contrib/docker/entrypoint.sh:58]: For now let's just add an annotation to allow the `eval` here, and then open an issue to fix the bug.

**Review by nuttycom — CHANGES_REQUESTED** [review]
The docker entrypoint change needs to be reverted & flagged off for now, otherwise utACK with @str4d

File: contrib/docker/entrypoint.sh (hunk 1 above)
Comment (sellout) [reply contrib/docker/entrypoint.sh]: https://github.com/zcash/zcash/issues/6297

**Review by nuttycom — APPROVED** [review]
utACK

**Pull request merged**